addons/GME/Scripts/Game/GME/Editor/Modules/GME_Modules_AmbientFlyby.c

```diff
@@ -0,0 +1,140 @@
+class GME_Modules_AmbientFlyby : GME_Modules_Base
+{
+	[Attribute(uiwidget: UIWidgets.ResourcePickerThumbnail, desc: "Prefab name for the ambient helicopter", params: "et")]
+	protected ResourceName m_sAmbientHeloPrefabName;
+	
+	[Attribute(defvalue: "50", desc: "Speed in m/s")]
+	protected float m_fSpeed;
+	
+	[Attribute(defvalue: "80", desc: "Flying height relative to terrain in meters")]
+	protected float m_fHeight;
+	
+	[Attribute(defvalue: "3000", desc: "Starting distance from player in meters")]
+	protected float m_fStartDistance;
+	
+	protected const float TERRAIN_Y_SCAN_STEP_SIZE = 100;
+	
+	protected Vehicle m_pAircraft;
+	protected GME_AmbientFlybyComponent m_pFlybyComponent;
+	
+	//------------------------------------------------------------------------------------------------
+	void SetSpeed(float speed)
+	{
+		m_fSpeed = speed;
+		
+		if (m_pFlybyComponent)
+			m_pFlybyComponent.SetSpeed(speed);
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	float GetSpeed()
+	{
+		return m_fSpeed;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	void SetHeight(float height)
+	{
+		m_fHeight = height;
+		
+		if (m_pFlybyComponent)
+		{
+			vector trasnform[4];
+			GetWorldTransform(trasnform);
+			OnMoved(trasnform, trasnform);
+		}
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	float GetHeight()
+	{
+		return m_fHeight;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	void SetStartDistance(float distance)
+	{
+		m_fStartDistance = distance;
+		
+		if (m_pFlybyComponent)
+		{
+			vector trasnform[4];
+			GetWorldTransform(trasnform);
+			OnMoved(trasnform, trasnform);
+		}
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	float GetStartDistance()
+	{
+		return m_fStartDistance;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	override void Run()
+	{
+		EntitySpawnParams params = new EntitySpawnParams();
+		params.TransformMode = ETransformMode.WORLD;
+		GetWorldTransform(params.Transform);
+		ComputeInitialAircraftTransform(params.Transform, params.Transform);
+		m_pAircraft = Vehicle.Cast(GetGame().SpawnEntityPrefab(Resource.Load(m_sAmbientHeloPrefabName), null, params));
+		m_pFlybyComponent = GME_AmbientFlybyComponent.Cast(m_pAircraft.FindComponent(GME_AmbientFlybyComponent));
+		m_pFlybyComponent.SetSpeed(m_fSpeed);
+		m_pFlybyComponent.SetDespawnDistance(2 * m_fStartDistance);
+		m_pFlybyComponent.GetOnDespawn().Insert(OnAircraftDespawned);
+		SCR_EditableEntityComponent editComponent = SCR_EditableEntityComponent.Cast(m_pAircraft.FindComponent(SCR_EditableEntityComponent));
+		editComponent.GetOnDeleted().Insert(OnAircraftDespawned);
```

Comment:
You don't need to null check it? I am not sure if that has changed, but I seem to recall initially we needed to null-check all find components before using them.

Reply:
An issue with doing null checks on components that should always be present is that such exceptions will go silent that way. I initially thought not checking them will at least throw an exception on the next line, but in hindsight it's bad practice. I think the best course of action is to do the null checks and in future provide an assertive cast method (I think Dedmen posted one on ACE Discord a while ago) that we can use in such cases.